### ISSUE TITLE
When quickly navigating, All Items occasionally treated as Trash and permanently deleted

### ISSUE THREAD

**Issue by ZacharyTalis** [bitwarden/web#1355]
### Steps To Reproduce

1. Have a huge amount of items which could take a while to load.
2. Login and directly click on the Trash link while the spinner is still active for loading all the vault items.

When deleting an individual item it will be sent to the Trash. However, when selecting multiple items and using the top-gear to `Permanently` delete these items, they will be **Permanently deleted**.

### Expected Result

Non-trashed items aren't permanently deleted and are instead loaded to trash.

Ideally, the Web Vault avoids such a 'multi-environment' state.

### Actual Result

Web Vault changes the interface for the Trash environment, which triggers a different API for multiple selections.

### Screenshots or Videos

![The bug reproduced. Both `All Items` and `Trash` are highlighted.](https://user-images.githubusercontent.com/23324896/146844165-0680878d-d3a0-4576-ae20-57b76b91fc7b.png)

https://user-images.githubusercontent.com/23324896/146945567-6d23ec3b-40c1-4f32-bbb8-824a33c3fe6b.mp4

### Additional Context

[Reposted from here.](https://github.com/dani-garcia/vaultwarden/issues/2159#issuecomment-998860808) Thread contains context on how we narrowed down the issue.

Issue appears identically between Vaultwarden and Self-Hosted Bitwarden configurations, and has been reproduced by two different users.

Let me know if you need any other specifics <3

### Operating System

Linux

### Web Browser

Firefox

### Browser Version

Developer Edition v96.0b7

### Build Version

2.25.0

**Comment by sso-bitwarden** [comment]
https://user-images.githubusercontent.com/75601583/161895157-2383ed86-407e-4ce7-bd60-3f825ce87272.mp4

**Comment by ZacharyTalis** [comment]
@sso-bitwarden yes, I understand the confirmation steps in place to prevent accidental deletions. Even if this issue ends up WONTFIX, I still feel it's valuable to make the bug known.

**Comment by sso-bitwarden** [comment]
@ZacharyTalis  I actually posted the video to bring more evidence :)

**Comment by ZacharyTalis** [comment]
@sso-bitwarden ohhh I'm a buffoon!! tysm, I thought you were suggesting I didn't know there was a Trash bin :D

**Comment by sso-bitwarden** [comment]
I was planning to poke our dev team after I posted that video :)

**Comment by ZacharyTalis** [comment]
Dang my quick response time, I missed out on snagging some context 😭

**Comment by SharkyRawr** [comment]
This happened to me too, exactly like the video clip shows. Was cleaning up my vault and accidentally deleted 500 items - which I luckily was able to restore from a backup.

**Comment by luislobo** [comment]
Same thing has happened to me, was migrating from Lastpass, and after having several imports, items were duplicated, deleted the duplicates, went to the trash, permanently deleted them. And suddenly I had them back on the Trash (or that was what I thought...) so, "Permanently deleted them again, as I thought, I might have not clicked it? or there was a pagination issue? Ended up with my Vault empty.

And I'm evaluating this to deploy to 100 enterprise users... I'll pause the purchase of the licenses until this gets resolved... And if it doesn't get resolved in a reasonable timeframe, I will have to use LastPass. Not that we prefer their service, but this issue is serious for us.
